src/detectors.rs

```diff
@@ -0,0 +1,340 @@
+    /// ```
+    pub fn new_single_threaded(interval: Duration, detection_time: Duration) -> (Self, Builder) {
+        LongRunningTaskDetector::new(interval, detection_time, true)
+    }
+
+    /// Creates a new instance of `LongRunningTaskDetector` linked to a multi-threaded Tokio runtime.
+    ///
+    /// This function takes the `interval` and `detection_time` parameters and initializes a
+    /// `LongRunningTaskDetector` with a multi-threaded Tokio runtime.
+    ///
+    /// # Parameters
+    ///
+    /// - `interval`: The time interval between probes.
+    /// - `detection_time`: The maximum blocking time allowed for detecting a long-running task.
+    ///
+    /// # Returns
+    ///
+    /// Returns a tuple containing the created `LongRunningTaskDetector` instance and the Tokio
+    /// runtime `Builder` used for configuration.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use tokio_metrics::detectors::LongRunningTaskDetector;
+    /// use std::time::Duration;
+    ///
+    /// let (detector, builder) = LongRunningTaskDetector::new_multi_threaded(Duration::from_secs(1), Duration::from_secs(5));
+    /// ```    
+    pub fn new_multi_threaded(interval: Duration, detection_time: Duration) -> (Self, Builder) {
+        LongRunningTaskDetector::new(interval, detection_time, false)
+    }
+
+    /// Starts the monitoring thread with default action handlers (that write details to std err).
+    ///
+    /// # Parameters
+    ///
+    /// - `runtime` - An `Arc` reference to a `tokio::runtime::Runtime`.    
+    pub fn start(&self, runtime: Arc<Runtime>) {
+        self.start_with_custom_action(runtime, Arc::new(StdErrBlockingActionHandler))
+    }
+
+    /// Starts the monitoring process with custom action handlers that
+    ///  allow you to customize what happens when blocking is detected.
+    ///
+    /// # Parameters
+    ///
+    /// - `runtime` - An `Arc` reference to a `tokio::runtime::Runtime`.
+    /// - `action` - An `Arc` reference to a custom `BlockingActionHandler`.
+    /// - `thread_action` - An `Arc` reference to a custom `ThreadStateHandler`.
+    pub fn start_with_custom_action(
+        &self,
+        runtime: Arc<Runtime>,
+        action: Arc<dyn BlockingActionHandler>,
+    ) {
+        *self.stop_flag.lock().unwrap() = false;
+        let stop_flag = Arc::clone(&self.stop_flag);
+        let detection_time = self.detection_time;
+        let interval = self.interval;
+        let workers = Arc::clone(&self.workers);
+        thread::spawn(move || {
+            let mut rng = thread_rng();
+            while !*stop_flag.lock().unwrap() {
+                probe(&runtime, detection_time, &workers, &action);
+                thread::sleep(Duration::from_micros(
+                    rng.gen_range(10..=interval.as_micros().try_into().unwrap()),
+                ));
+            }
+        });
+    }
+
+    /// Stops the monitoring thread. Does nothing if LRTD is already stopped.
```

Comment:
The LRTD contraction here is confusing. Why not say "monitoring thread" instead?